fe/fe-core/src/test/java/com/starrocks/analysis/SelectStmtWithCaseWhenTest.java

```diff
@@ -0,0 +1,553 @@
+                        "Predicates: 1: region NOT IN ('Japan', 'China')",
+                        "Predicates: 1: region NOT IN ('China', 'Japan')",
+                },
+                {"select * from test.t0 where \n" +
+                        "(case region when 'China' then 1 when 'Japan' then 2 else 3 end) not in (3)",
+                        "Predicates: 1: region IN ('Japan', 'China')",
+                        "Predicates: 1: region IN ('China', 'Japan')",
+                },
+
+                {"select * from test.t0 where \n" +
+                        "(case region when 'China' then 1 when 'Japan' then 2 else 3 end) is null",
+                        "  0:EMPTYSET\n" +
+                                "     cardinality: 1",
+                },
+                {"select * from test.t0 where \n" +
+                        "(case region when 'China' then 1 when 'Japan' then 2 else 3 end) is not null",
+                        "  0:OlapScanNode\n" +
+                                "     table: t0, rollup: t0\n" +
+                                "     preAggregation: on\n" +
+                                "     partitionsRatio=0/3, tabletsRatio=0/0\n" +
+                                "     tabletList=\n" +
+                                "     actualRows=0, avgRowSize=4.0\n" +
+                                "     cardinality: 1",
+                },
+
+                {"select * from test.t0 where \n" +
+                        "(case region when 'China' then 1 when 'Japan' then 2 end) is null",
+                        "Predicates: 1: region NOT IN ('Japan', 'China')",
+                        "Predicates: 1: region NOT IN ('China', 'Japan')",
+                },
+                {"select * from test.t0 where \n" +
+                        "(case region when 'China' then 1 when 'Japan' then 2 end) is not null",
+                        "Predicates: 1: region IN ('Japan', 'China')",
+                        "Predicates: 1: region IN ('China', 'Japan')",
+                },
+
+                {"select * from test.t0 where \n" +
+                        "(case region when 'China' then NULL when 'Japan' then 2 end) is null",
+                        "Predicates: (1: region = 'China') OR (1: region NOT IN ('China', 'Japan'))",
+                        "Predicates: (1: region = 'China') OR (1: region NOT IN ('Japan', 'China'))",
+                },
+                {"select * from test.t0 where \n" +
+                        "(case region when 'China' then 1 when 'Japan' then NULL end) is not null",
+                        "Predicates: [1: region, VARCHAR, false] != 'Japan', 1: region IN ('Japan', 'China')",
+                        "Predicates: [1: region, VARCHAR, false] != 'Japan', 1: region IN ('China', 'Japan')",
+                }
+        };
+        int i = 0;
+        for (String[] tc : testCases) {
+            if (i < 0) {
+                ++i;
```

Comment:
What does `i < 0` mean?  Maybe use `@ParameterizedTest` provided by `Junit5` is better. It can split cases in these arrays to separate unit tests.

Reply:
it can be removed